components/list/demo/list-demo-control.js

```diff
@@ -111,12 +131,28 @@ class ListDemoControl extends LitElement {
 		});
 	}
 
+	_onListChange(event) {
+		const list = event.target;
+		const selectAll = this.shadowRoot.querySelector('.select-all');
+		this._numberSelected = event.detail.selected.length;
+		selectAll.indeterminate = list.getSelectionState() === selectableListStates.indeterminate;
+		selectAll.checked = list.getSelectionState() === selectableListStates.all;
+
+		// This line allows you to see how and when the events fire. So check your console log.
+		console.log(event.detail.selected); // eslint-disable-line
```

Comment:
I keep adding this event to see how the events are triggering.
This information is great in the demo.